addons/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/blob/s3/S3BlobProvider.java

```diff
@@ -179,4 +189,45 @@ protected String getContentDispositionHeader(Blob blob) {
         return RFC2231.encodeContentDisposition(blob.getFilename(), false, null);
     }
 
+    @Override
+    public BlobStatus getStatus(ManagedBlob blob) throws IOException {
+        String key = stripBlobKeyPrefix(blob.getKey());
+        String objectKey;
+        String versionId;
+        int seppos = key.indexOf(VER_SEP);
+        if (seppos < 0) {
+            objectKey = key;
+            versionId = null;
+        } else {
+            objectKey = key.substring(0, seppos);
+            versionId = key.substring(seppos + 1);
+        }
+        String bucketKey = config.bucketPrefix + objectKey;
+        GetObjectMetadataRequest request = new GetObjectMetadataRequest(config.bucketName, bucketKey);
+        request.setVersionId(versionId);
+        ObjectMetadata metadata;
+        try {
+            metadata = config.amazonS3.getObjectMetadata(request);
+        } catch (AmazonServiceException e) {
+            if (S3BlobStore.isMissingKey(e)) {
+                // don't crash for a missing blob, even though it means the storage is corrupted
+                log.error("Failed to get information on blob: " + key, e);
+                return new BlobStatus();
+            }
+            throw e;
+        }
+        // storage class is null for STANDARD
+        String storageClass = defaultIfEmpty(metadata.getStorageClass(), StorageClass.Standard.toString());
+        // x-amz-restore absent
```

Comment:
i checkout you branch and i get an error compilation here, pehaps i miss something, but your static import 
`import static org.apache.commons.lang.StringUtils.defaultIfEmpty;` doesn't exist i replace it by `import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;` i think we use apache common 3

Reply:
Thanks. I intended to use commons-lang3 but I still have indirect dependencies to commons-lang somewhere.